synapse/http/client.py

```diff
@@ -306,6 +309,33 @@ def get_file(self, url, output_stream, max_size=None):
         defer.returnValue((length, headers, response.request.absoluteURI, response.code))
 
 
+class MatrixProxyClient(object):
+    """
+    An HTTP client that proxies other Matrix endpoints, ie. if the remote endpoint
+    returns Matrix-style error response, this will raise the appropriate SynapseError
```

Comment:
I'm a bit confused - why can't we just pass the Matrix-style error response through to the client?